flash.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import re
+
+def downloadESPtool():
+    output = subprocess.run(["sudo", "apt", "install", "esptool"], text=True)
+    if(output.returncode != 0):
+        raise Exeption(output.returncode)
+        
+def getNumber(maxInput):
+    num = 0
+    while True:
+        try:
+            num = int(input())
+            if (num >= 0 and num < maxInput):
+                break;
+            else: 
+                raise Exception("Wrong input number")
+        except:
+            print("Print a correct number.")
+    return num
+    
+def getCOMPortList():
+    ports = os.listdir("/dev/")
+    output = []
+    for i in range(len(ports) - 1):
+        if((re.search("ttyUSB*", ports[i]) != None) or (re.search("ttyACM*", ports[i]) != None)):
```

Comment:
1) No parentheses needed.
2) Comparison to None should be 'if cond is not None:'